### PR TITLE
React to product-owners.yml changes

### PR DESCRIPTION
**Pull request by chadwhitacre**
https://github.com/getsentry/security-as-code/blob/main/rbac/lib/product-owners.yml is the new source of truth for product areas. We want area labels in this repo to come from there, this gives us a workflow we can poke from over there to update the labels.

### CODE REVIEW

File: .github/labels.yml
Comment (hubertdeng123) [inline .github/labels.yml:211]: can this be single quotes for consistency purposes?

Comment (chadwhitacre) [reply .github/labels.yml:211]: Well, the value has a single quote. Are you asking to handle that in a different way?

Comment (hubertdeng123) [reply .github/labels.yml:211]: Ah, didn't notice the difference between `’` and `'` 🔬

File: bin/react-to-product-owners-yml-changes.py
Comment (hubertdeng123) [inline bin/react-to-product-owners-yml-changes.py:37]: 👀 Ah I see

File: .github/workflows/react-to-product-area-change.yml
Comment (hubertdeng123) [inline .github/workflows/react-to-product-area-change.yml]: Do the PR's still require approval? If yes, does this do anything?

**Comment by chadwhitacre** [comment]
Here's an example PR created by running the sh script locally:

https://github.com/getsentry/sentry-docs/pull/6860

File: bin/react-to-product-owners-yml-changes.sh
Comment (hubertdeng123) [inline bin/react-to-product-owners-yml-changes.sh:15]: Does this PR require approval? If yes, does this do anything?

File: bin/react-to-product-owners-yml-changes.py
Comment (hubertdeng123) [inline bin/react-to-product-owners-yml-changes.py:24]: Took a while to understand what's going on here, can we include some comments for why `fastforward` has to be used? Also describing head as the parts of the file before the `Product Area` labels and tail as the parts of the file after the `Product Area` labels

File: bin/react-to-product-owners-yml-changes.sh
Comment (chadwhitacre) [reply bin/react-to-product-owners-yml-changes.sh:15]: Yes, it will require approval. This sets it up to auto-merge once it is approved.

File: bin/react-to-product-owners-yml-changes.py
Comment (chadwhitacre) [reply bin/react-to-product-owners-yml-changes.py:30]: a4f16c85aa2227e72b0e160a7d48af95ed8fb8c6

**Review by hubertdeng123 — APPROVED** [review]

**Pull request merged**
